jerry-core/jerry.c

```diff
@@ -1841,81 +1824,117 @@ snapshot_add_compiled_code (ecma_compiled_code_t *compiled_code_p, /**< compiled
   if ((snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG) > 0xffffu)
   {
     snapshot_error_occured = true;
-    return;
+    return 0;
   }
 
-  snapshot_last_compiled_code_offset = snapshot_buffer_write_offset;
+  uint16_t start_offset = (uint16_t) (snapshot_buffer_write_offset >> MEM_ALIGNMENT_LOG);
+  ecma_compiled_code_t *copied_compiled_code_p;
 
-  compiled_code_map_entry_t *new_entry;
-  new_entry = (compiled_code_map_entry_t *) mem_heap_alloc_block (sizeof (compiled_code_map_entry_t));
+  copied_compiled_code_p = (ecma_compiled_code_t *) (snapshot_buffer_p + snapshot_buffer_write_offset);
 
-  if (new_entry == NULL)
+  if (!(compiled_code_p->status_flags & CBC_CODE_FLAGS_FUNCTION))
```

Comment:
The issue that @franc0is also mentioned is that within the body of this `if` we know that we are actually not dealing with `ecma_compiled_code_t` but `re_compiled_code_t`. It would be much comprehendible (and even safer) to access whatever fields we access as fields of `re_compiled_code_t` (by casting `compiled_code_p`).

Moreover, it should be noted at the struct declarations that the first 3 fields _must_ always be kept in sync.